pkg/boot/align/align.go

```diff
@@ -0,0 +1,17 @@
+// Copyright 2015-2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package align
+
+import "os"
+
+var pageMask = uint(os.Getpagesize() - 1)
+
+func AlignUp(p uint) uint {
```

Comment:
personally I wish this were called AlignPageSize or something.

Reply:
That is better name. Let me change it that way -- kexec user space does not even have the "up" part, which is more confusing

Reply:
the existence of this package, the dependency from pkg/boot/kexec to pkg/boot/linux are both indications that something is still wrong.

pkg/boot/linux should _exclusively_ depend on pkg/boot/kexec in the ideal case. Perhaps that means the objects belong in kexec, but the KexecLoad belongs in pkg/boot/linux.

Reply:
"exclusively" may break without more refactoring to kexec memory pieces, as multi boot need it to load segments also.

One *possible* ideal state maybe:

`pkg/boot/linux`:  classic load (purg load, etc) -- collectively, linux boot loader.
`pkg/boot/kexec`: golang wrapper to the kexec sys calls -- meant for boot loaders to call them. e.g. `linux`, and `multiboot` will call `Load` to do classic load. 
                            and, at top level, `pkg/boot/linux.go` switches to call linux boot loader `pkg/boot/linux`. So file load may also move to linux boot loader `pkg/boot/linux`.
kexec memory may as well move to a separate pkg, to be under `pkg/boot/`, as it will also be used by specific boot loaders, apart from the `Load` wrapper to classic sys call in `kexec`.

But bottom line I think: there are often more than one ways to group things, as long as it serves our purpose for the moment and near future.  Pre-mature generalization can be controversial as you mentioned in other thread.

Reply:
When I said exclusively, I'm sorry, that was confusing. I meant that both multiboot and linux should depend on kexec, and never the other way around. Right now we have multiboot depending on kexec, but kexec depending on Linux.